include/CL/cl_ext_pocl.h

```diff
@@ -60,42 +60,42 @@ typedef CL_API_ENTRY cl_int
 
 #endif
 
-/* cl_pocl_pinned_buffers (experimental stage) */
+/* cl_ext_buffer_device_address (experimental stage) */
 
-#ifndef cl_pocl_pinned_buffers
-#define cl_pocl_pinned_buffers 1
-#define CL_POCL_PINNED_BUFFERS_EXTENSION_NAME "cl_exp_pinned_buffers"
+#ifndef cl_ext_buffer_device_address
+#define cl_ext_buffer_device_address 1
 
-/* TODO: We need also platform/runtime extension due to the new buffer
-   creation flags. */
+/* We need both a platform and device extension due to the new buffer
+   creation flag. */
 
-/* clCreateBuffer(): A new cl_mem_flag CL_MEM_PINNED:
+/* clCreateBuffer(): A new cl_mem_flag CL_MEM_DEVICE_ADDRESS:
 
-   This flag specifies that the buffer must be persistently allocated
-   in the device's physical memory for its lifetime. That is, the buffer's
-   device address will remain the same and the space is reserved until
-   the buffer is freed. The device-specific buffer content updates are
-   still performed by implicit or explicit buffer migrations performed by
-   the runtime or the client code. If any of the devices in the context
-   does not support pinning, an error (TO DEFINE) is returned.
+   This flag specifies that the buffer must have a single fixed address
+   for its lifetime, which should be unique at least across the devices
+   of the context, but not necessarily with the host (virtual memory) as
+   with SVM or USM. The flag might imply that the buffer is "pinned"
+   permanently to a device's memory, but might not be necessarily so,
```

Comment:
Will indirect buffer accesses from kernels work in case the buffers happen not to be pinned to devices’ memory? Maybe this extension needs a new clSetKernelExecInfo() parameter, `CL_KERNEL_EXEC_INFO_<something>_PTRS`, to let drivers know about indirectly accessed buffers?

Reply:
Yes, that might be the most convenient and perf. portable way. The original semantics of this extension included "pinned" which as a side effect ensured consistency also for indirect accesses, but I wanted to relax that for additional implementation flexibility.

Another question is multi-device random access. For that we can add a new property for clCreateBufferWithProperties() later. So the initial prototype would be defined to work for single-device allocations only.